### PR TITLE
fix: read receipts now match the spec

### PR DESCRIPTION
**Pull request by CalumMcCall**
### Issues

Read receipts were getting sent when they weren't supposed to, as the group conversation setting was being ignored if it contained a value.

### Solutions

Some simple unit tests were added to help ensure the code remains compliant in future.

#### Notes

I've assumed that the conversation data has its read receipt setting updated by the backend when both clients have read receipts enabled, as this is my understanding of the spec.

#### APK
[Download build #1012](http://10.10.124.11:8080/job/Pull%20Request%20Builder/1012/artifact/build/artifact/wire-dev-PR2396-1012.apk)
[Download build #1068](http://10.10.124.11:8080/job/Pull%20Request%20Builder/1068/artifact/build/artifact/wire-dev-PR2396-1068.apk)

### CODE REVIEW

**Comment by AndroidBob** [comment]
Can one of the admins verify this patch?

**Comment by johnxnguyen** [comment]
ok to test

File: wire-android-sync-engine/zmessaging/src/main/scala/com/waz/service/conversation/ConversationsUiService.scala [hunk 1]
```diff
@@ -396,19 +396,28 @@ class ConversationsUiServiceImpl(selfUserId:        UserId,
     _   <- sync.postMessage(msg.id, id, msg.editTime)
   } yield Some(msg)
 
+  def shouldSendReadReceipts(convId: ConvId, readReceiptSettings: ReadReceiptSettings): Future[Boolean] =
+    convs.isGroupConversation(convId).map {
+      case true =>
+        readReceiptSettings.convSetting.contains(1)
+      case false =>
+        readReceiptSettings.selfSettings && readReceiptSettings.convSetting.contains(1)
```
Comment (marcoconti83) [inline wire-android-sync-engine/zmessaging/src/main/scala/com/waz/service/conversation/ConversationsUiService.scala]: I have to double check on this as I'm not sure: `readReceiptSettings.convSetting.contains(1)`

In your PR comment you say: 
> I've assumed that the conversation data has its read receipt setting updated by the backend when both clients have read receipts enabled, as this is my understanding of the spec.

But it's possible that this does not happens for 1:1. Will keep this PR open until this is clarified, and if needed we will update it.

Comment (marcoconti83) [reply wire-android-sync-engine/zmessaging/src/main/scala/com/waz/service/conversation/ConversationsUiService.scala]: In 1:1 conversations, read receipt setting is computed to be true only if the two users have read receipt enabled. 
In group conversations, read receipt setting is a property of the conversation itself.

I think this difference in behavior is not captured by the code above.

**Comment by android10** [comment]
@makingthematrix @marcoconti83 do you know if we can either merge or close this one? AFAIK it is only adding tests.

**Comment by marcoconti83** [comment]
> @makingthematrix @marcoconti83 do you know if we can either merge or close this one? AFAIK it is only adding tests.

It also changes some behavior, it's not just tests. I didn't review @makingthematrix latest changes, I was hoping someone else from the team would review it. If the code is approved, then this is good to merge.

**Comment by android10** [comment]
@marcoconti83 thanks for the clarification. I do not have enough context on this one. Maybe @gizemb you can help us here?

**Comment by gizemb** [comment]
@marcoconti83 your prior change request is blocking the merge. can you have another look?

**Comment by marcoconti83** [comment]
Done!

**Pull request merged**